### DISCUSSION TITLE
Presence lock on conditions.

### DISCUSSION THREAD

**Discussion by myevit** [jseidl/magic-areas#446] [Ideas / Feature Requests]
Hi,
I think it will be useful to have presence lock on certain conditions. 
For example: I have an interior door sensor, when the door is closed and there is motion detected or light switched on/off (essentially any human activity) I want to lock presence on for this room till the door will be open.
Second example: On the way to the basement, I have a motion sensor, when it triggers on and then off and if in the basement there is any human activity detected - I want to lock the presence sensor on till that motion sensor will be triggered again, which reset release the presence lock and will wait for any activity in the basement again. 
this is applicable for whole house/apartment if there are door sensors. 

I hope it make sense.

p.s. Wasp in a Box I guess is the other name for it...

**Comment by jseidl** [comment]
Hey! Yes I was about to ask if maybe you were talking about Wasp in a box. There's an FR for that here https://github.com/jseidl/hass-magic_areas/discussions/226, please upvote that. It's close to being worked on btw!
